File: javascript/TECNICATURA3JS/TSLeccion07/Empleado.js
class Empleado extends Persona {
  static contadorEmpleado = 0;

  constructor(nombre, apellido, edad, sueldo) {
    super(nombre, apellido, edad);
    this._contadorEmpleado = ++Empleado.contadorPersonas;
    this._sueldo = sueldo;
  }

  get idEmpleado() {
    return this._contadorEmpleado;
  }

  get sueldo() {
    return this._sueldo;
  }

  set sueldo(sueldo) {
    this._sueldo = sueldo;
  }

  toString() {
    return `
    ${super.toString()}
    ${this._idEmpleado}
    ${this._sueldo}
    `;
  }
}

File: Guardini-Philama/JAVA/Clase_09/Herencia/src/domain/Empleado.java

package domain;

public class Empleado extends Persona{ //con extends, decimos de donde va a heredar la clase
    private int idEmpleado;
    private double sueldo;
    private static int contadorEmpleados;//es para incrementar

    //Constructores
    public Empleado(){ //Constructor 1
         this.idEmpleado = ++Empleado.contadorEmpleados;
    }
    
    public Empleado(String nombre, double sueldo) { //Constructor 2
        //super(nombre); //si usamos super, no podemos usar this para llamar a un constructor interno
        this(); //EStamos llamando desde aqui al constructor vacio (llamar a un constructor interno)
        this.nombre = nombre;
        this.sueldo = sueldo;
    }

    public int getIdEmpleado() {
        return this.idEmpleado;
    }

    public double getSueldo() {
        return this.sueldo;
    }

    public void setSueldo(double sueldo) {
        this.sueldo = sueldo;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Empleado{idEmpleado=").append(idEmpleado);
        sb.append(", sueldo=").append(sueldo);
        sb.append(", ").append(super.toString());
        sb.append('}');
        return sb.toString();
    }
}